cranelift/codegen/src/isa/aarch64/inst/emit.rs

```diff
@@ -954,7 +956,7 @@ impl MachInstEmit for Inst {
                 let (mem_insts, mem) = mem_finalize(Some(sink), &mem, state);
 
                 for inst in mem_insts.into_iter() {
-                    inst.emit(&[], sink, emit_info, state);
+                    inst.emit(&[], sink, emit_info, state, ctrl_plane);
```

Comment:
`ctrl_plane` can probably go inside the `state` (`EmitState`)?

The issue with lifetimes that this would otherwise create (`&mut ControlPlane` inside of the struct) can be resolved I think by `std::mem::move` to take ownership of the control plane temporarily in places where we emit.